dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/security/ImpersonateUserController.java

```diff
@@ -125,18 +125,37 @@ public ImpersonateUserResponse impersonateUser(
     }
   }
 
-  private void validateRequest(HttpServletRequest request, UserDetails userDetails)
+  private static void validateImpersonateAuthority(UserDetails userDetails)
+      throws ForbiddenException {
+    if (userDetails.isSuper()
+        || userDetails.getAllAuthorities().contains(F_IMPERSONATE_USER.name())) {
+      return;
+    }
+    throw new ForbiddenException("Forbidden, requires authority [F_IMPERSONATE_USER]");
+  }
+
+  private static void validateImpersonateExitAuthority(UserDetails userDetails)
+      throws ForbiddenException {
+    if (userDetails.isSuper()) {
+      return;
+    }
+    Collection<? extends GrantedAuthority> authorities = userDetails.getAuthorities();
+    for (GrantedAuthority authority : authorities) {
+      if (authority.getAuthority().equals(F_IMPERSONATE_USER.name())
+          || authority.getAuthority().equals(F_PREVIOUS_IMPERSONATOR_AUTHORITY.name())) {
+        return;
+      }
+    }
+    throw new ForbiddenException("Forbidden, requires authority [F_IMPERSONATE_USER]");
```

Comment:
looks like this method looks for either `F_IMPERSONATE_USER` or `F_PREVIOUS_IMPERSONATOR_AUTHORITY`, but only `F_IMPERSONATE_USER` is in the error response. Might be misleading for the user. Should `F_PREVIOUS_IMPERSONATOR_AUTHORITY` be included in the error response?

Reply:
Added!